src/version.c

```diff
@@ -10,32 +10,88 @@
 #include <funcapi.h>
 #include <fmgr.h>
 #include <storage/fd.h>
+#include <utils/timestamp.h>
 
 #include "fmgr.h"
 #include "compat.h"
+#include "annotations.h"
 #include "gitcommit.h"
 #include "version.h"
 #include "config.h"
 
-#define STR_EXPAND(x) #x
-#define STR(x) STR_EXPAND(x)
+/* Export the strings to that we can read them using strings(1). We add a
+ * prefix so that we can easily find it using grep(1). We only bother about
+ * generating them if the relevant symbol is defined. */
+#ifdef EXT_GIT_COMMIT_HASH
```

Comment:
Will this work as expected?

Even if `@EXT_GIT_COMMIT_TAG@` is empty in gitcommit.h.in, I suspect  `EXT_GIT_COMMIT_TAG` will still be defined (albeit empty)?

Reply:
Still unclear what happens if `EXT_GIT_COMMIT_HASH` is defined like this:

```
#define EXT_GIT_COMMIT_HASH
```

Reply:
But it is not defined like that in the generated file, it is either

```
#define EXT_GIT_COMMIT_HASH "2beeb3fc"
```

or 

```
/* #undef EXT_GIT_COMMIT_HASH */
```

If a builder decides on editing the file and putting anything else there, they can break the build whatever we do, so there is no reason to handle that case.

Reply:
That's what I wanted to know.